### ISSUE TITLE
Incorrect recognition of phone number

### ISSUE THREAD

**Issue by hunterwilhelm** [calcom/cal.com#15931]
### Issue Summary

This issue is related to #7370, however, that is a self-hosted cal, this bug is on cal.com directly.

When autofilling a 480 number without a country prefix, the default country shows US, but then it switches to a different country. This is not satisfactory for my needs. If there isn't a plus on the front, it should keep the +1 and append the autofilled phone number.

### Steps to Reproduce

1. Put 4805551234 into your autofill in chrome

2. Select that autofill

<img width="434" alt="Screenshot 2024-07-26 at 11 41 50 AM" src="https://github.com/user-attachments/assets/67cc4899-88a4-4974-a141-eba3c3214834">

3. Behold, a different country shows up.

<img width="435" alt="Screenshot 2024-07-26 at 11 41 56 AM" src="https://github.com/user-attachments/assets/8608bb86-008d-4283-9ca5-be5f74404f82">

### Expected behavior

I should keep the +1 in front of the autofilled phone number if that phone number doesn't have a plus.

### Environment

- [X] app.cal.com
- [ ] Self-hosted Cal

### Desktop (please complete the following information)

```markdown
- OS: MacOS
- Browser: Chrome
- Version: Version 126.0.6478.183 (Official Build) (arm64)
```

**Comment by ymohit1603** [comment]
@PeerRich can you assign me this issue buddy i can try my best to fix it.

**Comment by ciaranha** [comment]
@ymohit1603 no need to have an issue assigned.  Just open a pr and link back to this issue. Someone from the team will review when it’s ready.

**Comment by ymohit1603** [comment]
@ciaranha Ok i got it man thanks.

**Comment by hunterwilhelm** [comment]
Possibly fixed by #15991, I'll leave this open until I can confirm the autofill part of this issue is fixed. I don't have time to test this until it gets released later. I'll come back and update this issue when that happens.

**Comment by DEODENTALNICK** [comment]
We're having trouble with this as well. Is there a way to set a default country code in the account settings or something? Cal.com is almost unusable right now for us. Very few people enter their country code when putting in their phone number.

**Comment by hunterwilhelm** [comment]
I agree with @DEODENTALNICK, the autofill isn't working correctly. As it won't fill the country code even if it's in the browser settings.

v.4.3.5-h-3a46375

![Screenshot 2024-08-07 at 3 07 49 PM](https://github.com/user-attachments/assets/6c330f3c-3043-483f-bf93-04afe40fbce4)

Still defaults to a different country whose official languages don't even include English!
 
![Screenshot 2024-08-07 at 3 07 58 PM](https://github.com/user-attachments/assets/d2ef1b0e-0acb-4cf9-a3a3-a937893624c4)

I'll probably create a pull request soon about this.

**Comment by sarmadbokhari** [comment]
+1 still seeing this issue. Would love to see this bug fixed.

**Comment by edonv** [comment]
+1 this still seems to be an issue. Any update?
